### PR TITLE
Customize notifications list

### PR DESCRIPTION
**Pull request by prsvic**
The main goal of this PR is to allow more flexible styling of notification messages in `NotificationsList`.
This is achieved by introducing `renderNotificationMessage` property to `Notifications` component, which, if provided, will override default rendering function of a notification element.

Also some new style variables were exposed to the theme, for more detailed customization of widget.

### CODE REVIEW

File: packages/dialect-react-ui/components/Notifications/Header.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ function Header(props: HeaderProps) {
     <>
       <div className="dt-flex">
         <SettingsButton />
-        <div className={clsx(!props.settingsOnly && 'sm:dt-hidden', 'dt-ml-3')}>
```
Comment (fsher) [inline packages/dialect-react-ui/components/Notifications/Header.tsx:61]: This seems like a breaking change. Is this necessary to remove?

File: packages/dialect-react-ui/components/Notifications/screens/NotificationsList/index.tsx [hunk 2]
```diff
@@ -43,16 +51,22 @@ const NotificationsList = ({ refreshInterval }: NotificationsListProps) => {
...
-            timestamp={message.timestamp.getTime()}
-          />
-          <Divider className={notificationsDivider} />
-        </React.Fragment>
-      ))}
+    <div className={notificationsListWrapper}>
+      {messages.map((message, idx) =>
+        renderNotificationMessage ? (
+          <React.Fragment key={idx}>
+            {renderNotificationMessage(message)}
```
Comment (fsher) [inline packages/dialect-react-ui/components/Notifications/screens/NotificationsList/index.tsx:58]: I'd recommend passing the index as well.

**Review by fsher — APPROVED** [review]
Well done! Just cpl comments

File: packages/dialect-react-ui/components/Notifications/Header.tsx (hunk 1 above)
Comment (prsvic) [reply packages/dialect-react-ui/components/Notifications/Header.tsx:61]: added explicit prop to configure close button visibility

**Pull request merged**